### PR TITLE
Coalesce develop

### PR DESCRIPTION
**Pull request by spacerockzero**
Per-lang translation loader and file coalescer
link react-scripts in in a frontier react app to test

### CODE REVIEW

File: packages/react-scripts/per-lang-loader/PerLangPlugin.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        Object.keys(allLocales[locale]).forEach((namespace) => {
+          jsonfile.writeFileSync(`${path.join(builtLocalesDir, locale, namespace)}.json`, allLocales[locale][namespace])
+        })
+      })
+      callback()
+    });
+  }
+}
+
+module.exports = PerLangPlugin;
```
Comment (redbugz) [inline packages/react-scripts/per-lang-loader/PerLangPlugin.js:60]: should we commit this file if we don't use it?

File: packages/react-scripts/per-lang-loader/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Language-specific loading
```
Comment (redbugz) [inline packages/react-scripts/per-lang-loader/README.md:1]: at some point we need to fill this in

File: packages/react-scripts/per-lang-loader/PerLangPlugin.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+const builtLocalesDir = `${__dirname}/builtLocales`
+
+class PerLangPlugin {
+  apply(compiler) {
+    const allLocales = {};
+    const manifest = {};
+    // inject a function to loaderContext so loaders can pass back info
+    compiler.hooks.beforeCompile.tapAsync('PerLangPlugin', (params, callback) => {
+      debugger;
+      if (!fs.existsSync(builtLocalesDir)) {
```
Comment (redbugz) [inline packages/react-scripts/per-lang-loader/PerLangPlugin.js:17]: we're in tapAsync but making sync calls. for manually invoked scrips, sync calls are okay, but for build steps that might be run in parallel, we should probably do this properly async

File: packages/react-scripts/per-lang-loader/PerLangPlugin.js (hunk 1 above)
Comment (spacerockzero) [reply packages/react-scripts/per-lang-loader/PerLangPlugin.js:60]: I was seeing when @tylergraf was going to change the code to the other strategy, so I left it for him

File: packages/react-scripts/config/webpack.config.js [hunk 4]
```diff
@@ -425,10 +425,13 @@ module.exports = function (webpackEnv) {
             },
             // load locale files
             {
-              test: /locales/,
-              loader: '@alienfast/i18next-loader',
+              test: /locales\/index\.js$/,
+              loader: isEnvProduction
+                ? require.resolve('../per-lang-loader')
```
Comment (spacerockzero) [inline packages/react-scripts/config/webpack.config.js:430]: @redbugz it's currently pulled in here, though @tylergraf said he was changing that strategy to something else

Comment (spacerockzero) [reply packages/react-scripts/config/webpack.config.js:430]: nvm, I see this is just getting index.js

File: packages/react-scripts/per-lang-loader/PerLangPlugin.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+const builtLocalesDir = `${__dirname}/builtLocales`
+
+class PerLangPlugin {
+  apply(compiler) {
+    const allLocales = {};
+    const manifest = {};
+    // inject a function to loaderContext so loaders can pass back info
+    compiler.hooks.beforeCompile.tapAsync('PerLangPlugin', (params, callback) => {
+      debugger;
```
Comment (spacerockzero) [inline packages/react-scripts/per-lang-loader/PerLangPlugin.js:16]: reminder: delete me

**Comment by spacerockzero** [comment]
Note:
cra template gitignore contains `/dist` and not `dist`, so most apps that haven't deviated *will* see these generated locale files in their src and unstaged diffs if they run `build` locally
https://github.com/fs-webdev/cra-template/blob/master/template/gitignore#L15

File: packages/react-scripts/per-lang-loader/PerLangPlugin.js (hunk 3 above)
Comment (spacerockzero) [reply packages/react-scripts/per-lang-loader/PerLangPlugin.js]: file is deleted now

**Comment by joeycozza** [comment]
closing this PR in favor of the "real-deal" PR as per Tyler's instructions.

**Pull request closed without merging**